FTBFS with Qt 6.9.0

libqtxdg FTBFS against the tip of qtbase upstream with the following error:

```
In file included from /root/libqtxdg/src/xdgiconloader/xdgiconloader.cpp:37:
/root/libqtxdg/src/xdgiconloader/xdgiconloader_p.h:64:8: error: cannot derive from ‘final’ base ‘ScalableEntry’ in derived type ‘ScalableFollowsColorEntry’
   64 | struct ScalableFollowsColorEntry : public ScalableEntry
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~
```

This is due to https://code.qt.io/cgit/qt/qtbase.git/commit/?id=aa38df53b977d8c04123f8c0994d041d183a7971

At the top of the header file there is:
```
//
//  W A R N I N G
//  -------------
//
// This file is not part of the Qt API.  It exists purely as an
// implementation detail.  This header file may change from version to
// version without notice, or even be removed.
//
// We mean it.
//
```

I'm not exactly sure how to address this.

--- a/src/xdgiconloader/xdgiconloader.cpp
+++ b/src/xdgiconloader/xdgiconloader.cpp
@@ -439,7 +439,11 @@ QThemeIconInfo XdgIconLoader::findIconHelper(const QString &themeName,
                 } else if (gSupportsSvg) {
                     const QString svgPath = subDir + svgIconName;
                     if (QFile::exists(svgPath)) {
-                        std::unique_ptr<ScalableEntry> iconEntry = (followColorScheme() && theme.followsColorScheme()) ? std::make_unique<ScalableFollowsColorEntry>() : std::make_unique<ScalableEntry>();
+                        std::unique_ptr<QIconLoaderEngineEntry> iconEntry;
+                        if (followColorScheme() && theme.followsColorScheme())
+                            iconEntry.reset(new ScalableFollowsColorEntry);
+                        else
+                            iconEntry.reset(new ScalableEntry);
                         iconEntry->dir = dirInfo;
                         iconEntry->filename = svgPath;
                         info.entries.push_back(std::move(iconEntry));

--- a/src/xdgiconloader/xdgiconloader.cpp
+++ b/src/xdgiconloader/xdgiconloader.cpp
@@ -741,8 +741,11 @@ QSize XdgIconLoaderEngine::actualSize(const QSize &size, QIcon::Mode mode,
     QIconLoaderEngineEntry *entry = entryForSize(m_info, size);
     if (entry) {
         const QIconDirInfo &dir = entry->dir;
-        if (dir.type == QIconDirInfo::Scalable || dynamic_cast<ScalableEntry *>(entry))
+        if (dir.type == QIconDirInfo::Scalable
+            || dynamic_cast<ScalableEntry *>(entry)
+            || dynamic_cast<ScalableFollowsColorEntry *>(entry)) {
             return size;
+        }
         else {
             int dir_size = dir.size;
             //Note: fallback for directories that don't have its content size defined

--- a/src/xdgiconloader/xdgiconloader_p.h
+++ b/src/xdgiconloader/xdgiconloader_p.h
@@ -61,13 +61,14 @@
 
 class XdgIconLoader;
 
-struct ScalableFollowsColorEntry : public ScalableEntry
+struct ScalableFollowsColorEntry : public QIconLoaderEngineEntry
 {
 #if (QT_VERSION >= QT_VERSION_CHECK(6,8,0))
     QPixmap pixmap(const QSize &size, QIcon::Mode mode, QIcon::State state, qreal scale) override;
 #else
     QPixmap pixmap(const QSize &size, QIcon::Mode mode, QIcon::State state) override;
 #endif
+    QIcon svgIcon;
 };
 
 //class QIconLoaderEngine : public QIconEngine

